ddtrace/internal/datadog/profiling/stack_v2/CMakeLists.txt

```diff
@@ -0,0 +1,117 @@
+cmake_minimum_required(VERSION 3.19)
+
+# We're building a Python extension and expect to be called from setup.py
+set(EXTENSION_NAME "" CACHE STRING "Name of the extension")
+project(${EXTENSION_NAME})
+
+# Set C++17 as required standard for the whole project.
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Wextra")
+
+# Features
+option(UNWIND_NATIVE_ENABLE "Enable native unwinding" OFF)
+
+# Includes
+include(FetchContent)
+include(ExternalProject)
+include(FindPython3)
+add_subdirectory(../dd_wrapper "${LIB_INSTALL_DIR}")
+
+# Find the Python interpreter
+find_package(Python3 COMPONENTS Interpreter REQUIRED)
+if (NOT Python3_INCLUDE_DIRS)
+  message(FATAL_ERROR "Python3_INCLUDE_DIRS not found")
+endif()
+
+# Add echion
+FetchContent_Declare(
+    echion
+    GIT_REPOSITORY "https://github.com/sanchda/echion.git"
```

Comment:
Could we fetch a prebuilt artifact from some internal repository perhaps?

Reply:
Right now this work is based on a fork with several modifications--the official location of echion is @P403n1x87's repo.  At some point, we'll probably hash out whether/how to incorporate the spirit of these changes into echion (and if not, how to extend what echion is willing to provide with our own code).  From that point forward, I believe @P403n1x87's repo will be the official source.

Compiling echion is super fast compared to many of the other dependencies in dd-trace-py, and having source availability at this point is very useful as we have several design questions looming about how native dependencies will work in dd-trace-py.  I'm reluctant to prebuild juuust yet.

I think the prevailing concern is supply-chain, since @P403n1x87 is neither Mr. Datadog nor Mr. Pypi (citation needed).

I'll add a patch that will pull down the specified tag, but fail the build if the included sources and headers don't match pre-populated hashes.  I don't think keeping this up-to-date will be very onerous and unless I'm missing something, won't be inferior to the implicit trust we place in other package managers in our build process.